mkosi/qemu.py

```diff
@@ -161,14 +157,14 @@ def start_swtpm() -> Iterator[Path]:
 
 
 @contextlib.contextmanager
-def start_virtiofsd(directory: Path) -> Iterator[Path]:
-    uid, gid = InvokingUser.uid_gid()
-
+def start_virtiofsd(directory: Path, uid: Optional[int] = None, gid: Optional[int] = None) -> Iterator[Path]:
     with tempfile.TemporaryDirectory() as state:
         # Make sure virtiofsd is allowed to create its socket in this temporary directory.
-        os.chown(state, uid, gid)
+        os.chown(state, uid if uid is not None else os.getuid(), gid if gid is not None else os.getgid())
 
-        sock = Path(state) / Path("sock")
+        # Make sure we can use the socket name as a unique identifier for the fs as well but make sure it's not too
+        # long as virtiofs tag names are limited to 36 bytes.
+        sock = Path(state) / f"sock-{uuid.uuid4().hex[:30]}"
```

Comment:
```suggestion
        sock = Path(state) / f"sock-{uuid.uuid4().hex}"[:35]
```
less mental arithmetic